To: vim_dev@googlegroups.com
Subject: Patch 7.3.278
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.278
Problem:    Passing the file name to open in VisVim doesn't work.
Solution:   Adjust the index and check for end of buffer. (Jiri Sedlak)
Files:	    src/VisVim/Commands.cpp


*** ../vim-7.3.277/src/VisVim/Commands.cpp	2010-08-15 21:57:27.000000000 +0200
--- src/VisVim/Commands.cpp	2011-08-10 17:21:27.000000000 +0200
***************
*** 549,555 ****
  	if (g_bNewTabs)
  	{
  		sprintf(VimCmd, ":tab drop ");
! 		s = VimCmd + 11;
  	}
  	else
  	{
--- 549,555 ----
  	if (g_bNewTabs)
  	{
  		sprintf(VimCmd, ":tab drop ");
! 		s = VimCmd + 10;
  	}
  	else
  	{
***************
*** 557,564 ****
  		s = VimCmd + 6;
  	}
  	sprintf(FileNameTmp, "%S", (char *)FileName);
! 	for (p = FileNameTmp; *p != '\0' && s < FileNameTmp + MAX_OLE_STR - 4;
! 									  ++p)
  		if (*p == '\\')
  			*s++ = '/';
  		else
--- 557,563 ----
  		s = VimCmd + 6;
  	}
  	sprintf(FileNameTmp, "%S", (char *)FileName);
! 	for (p = FileNameTmp; *p != '\0' && s < VimCmd + MAX_OLE_STR - 4; ++p)
  		if (*p == '\\')
  			*s++ = '/';
  		else
*** ../vim-7.3.277/src/version.c	2011-08-10 17:07:56.000000000 +0200
--- src/version.c	2011-08-10 17:25:20.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     278,
  /**/

-- 
ARTHUR:          But if he was dying, he wouldn't bother to carve
                 "Aaaaarrrrrrggghhh".  He'd just say it.
BROTHER MAYNARD: It's down there carved in stone.
GALAHAD:         Perhaps he was dictating.
                 "Monty Python and the Holy Grail" PYTHON (MONTY) PICTURES LTD

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
